src/scenic/syntax/compiler.py

```diff
@@ -94,8 +215,150 @@ def visit_Name(self, node: ast.Name) -> any:
         # TODO(shun): Add handling for behavior locals
         return node
 
+    # hook control flow nodes to set appropriate flags for try-interrupt
+
+    def visit_For(self, node):
+        old = self.inLoop
+        self.inLoop = True
+        newNode = self.generic_visit(node)
+        self.inLoop = old
+        return newNode
+
+    def visit_While(self, node):
+        old = self.inLoop
+        self.inLoop = True
+        newNode = self.generic_visit(node)
+        self.inLoop = old
+        return newNode
+
+    def visit_FunctionDef(self, node):
+        oldInLoop, oldInInterruptBlock = self.inLoop, self.inInterruptBlock
+        self.inLoop, self.inInterruptBlock = False, False
+        newNode = self.generic_visit(node)
+        self.inLoop, self.inInterruptBlock = oldInLoop, oldInInterruptBlock
+        return newNode
+
+    def visit_Break(self, node):
+        if self.inInterruptBlock and not self.inLoop:
+            if not self.usedBreak:
+                self.usedBreak = node
+            newNode = ast.Return(breakFlag)
+            return ast.copy_location(newNode, node)
+        else:
+            return self.generic_visit(node)
+
+    def visit_Continue(self, node):
+        if self.inInterruptBlock and not self.inLoop:
+            if not self.usedContinue:
+                self.usedContinue = node
+            newNode = ast.Return(continueFlag)
+            return ast.copy_location(newNode, node)
+        else:
+            return self.generic_visit(node)
+
+    def visit_Return(self, node):
+        if self.inInterruptBlock:
+            value = ast.Constant(None) if node.value is None else node.value
+            ret = ast.Return(ast.Call(returnFlag, [value], []))
+            return ast.copy_location(ret, node)
+        else:
+            return self.generic_visit(node)
+
     # Special Case
 
+    def visit_ScenicTry(self, node: s.ScenicTry):
+        statements = []
+        oldInTryInterrupt = self.inTryInterrupt
+
+        # TODO(shun): Do we need to create a dead copy? https://github.com/BerkeleyLearnVerify/Scenic/blob/eb7e7f7ad1879c70f0f6d9f5cff7b83afd23ef7a/src/scenic/syntax/translator.py#L1707-L1709
```

Comment:
You're right, this is no longer required: I'd forgotten that we now implement local variables of behaviors and modular scenarios as attributes (so that one can write `myScenario.someObject` for example; as a side effect this syntax also works for behaviors). I've removed the dead copy from the `main` branch and added tests for the cases that it used to be necessary for, just in case (in commit e0f3f0c0).

Reply:
Thanks for your investigation. Removed the comment in b65ffdb19489db98da9b53d756d0531c913204cf